gourmet/tests/test_reccard.py

```diff
@@ -1,43 +1,62 @@
-from . import old_test  # get ../lib/ in path
-from gi.repository import Gtk
+from tempfile import TemporaryDirectory
+
+import gi
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk  # noqa: import not a top of file
+
+from gourmet import convert, gglobals  # noqa: import not at top
+from gourmet.backends.db import RecData  # noqa: import not at top
+from gourmet.reccard import add_with_undo, RecCard, RecCardDisplay  # noqa
 
-from gourmet import gglobals
-gglobals.gourmetdir = '/tmp/'
-gglobals.dbargs['file'] = '/tmp/recipes.db'
 
 VERBOSE = True
 
-from gourmet import GourmetRecipeManager
-from gourmet.reccard import add_with_undo
 
-def assert_with_message (callable,
-                         description):
+def print_(*msg):
+    if VERBOSE:
+        print(*msg)
+
+
+def assert_with_message(callable_, description):
     try:
-        assert(callable())
+        assert(callable_())
     except AssertionError:
-        print('FAILED:',description)
+        print('FAILED:', description)
         raise
     else:
         if VERBOSE:
-            print('SUCCEEDED:',description)
+            print('SUCCEEDED:', description)
+
 
-def add_save_and_check (rc, lines_groups_and_dc):
+def add_save_and_check(rc, lines_groups_and_dc):
+    idx = rc.recipe_editor.module_tab_by_name["ingredients"]
+    ing_controller = rc.recipe_editor.modules[idx].ingtree_ui.ingController
+
+    # All ingredient addition actions pass through add_with_undo
+    ing_editor = ing_controller.ingredient_editor_module
     added = []
-    for l,g,dc in lines_groups_and_dc:
-        # add_with_undo is what's called by any of the ways a user can add an ingredient.
+    for line, group, desc in lines_groups_and_dc:
         add_with_undo(
```

Comment:
If I run just this module locally, the current failure is here.

```
  File "/home/takluyver/Code/gourmet/gourmet/tests/test_reccard.py", line 300, in <module>
    test_ingredients_editing(rec_card)
  File "/home/takluyver/Code/gourmet/gourmet/tests/test_reccard.py", line 90, in test_ingredients_editing
    add_save_and_check(
  File "/home/takluyver/Code/gourmet/gourmet/tests/test_reccard.py", line 39, in add_save_and_check
    add_with_undo(
  File "/home/takluyver/Code/gourmet/gourmet/reccard.py", line 2838, in add_with_undo
    idx = rc.re.module_tab_by_name["ingredients"]
AttributeError: 'RecCard' object has no attribute 're'
```

Reply:
Yes, it should be `rc.recipe_editor`. I suppose that I have uncommitted changes

Reply:
So, following PR #130, it turns out that in the test, I pass a `RecCard` object, whereas an `IngredientEditorModule` is expected.